main.go

```diff
@@ -105,66 +104,74 @@ func run(ctx context.Context, wg *sync.WaitGroup, globalCollectionOpts state.Col
 	if globalCollectionOpts.TestRun {
-		if globalCollectionOpts.TestReport != "" {
-			runner.RunTestReport(servers, globalCollectionOpts, logger)
-			return
-		} else if globalCollectionOpts.TestExplain {
-			for _, server := range servers {
-				prefixedLogger := logger.WithPrefix(server.Config.SectionName)
-				err := runner.EmitTestExplain(server, globalCollectionOpts, prefixedLogger)
-				if err != nil {
-					prefixedLogger.PrintError("Failed to run test explain: %s", err)
-				}
-			}
-			return
-		} else if globalCollectionOpts.TestRunLogs {
-			reloadOkay = doLogTest(servers, globalCollectionOpts, logger)
-			return
-		} else {
-			var allFullSuccessful bool
-			var allActivitySuccessful bool
-			allFullSuccessful = runner.CollectAllServers(servers, globalCollectionOpts, logger)
-			if hasAnyActivityEnabled {
-				allActivitySuccessful = runner.CollectActivityFromAllServers(servers, globalCollectionOpts, logger)
-			} else {
-				allActivitySuccessful = true
-			}
-			if hasAnyLogsEnabled {
-				// We intentionally don't fail for the regular test command if the log test fails, since you may not
-				// have Log Insights enabled on your plan (which would fail the log test when getting the log grant).
-				// In these situations we still want --test to be successful (i.e. issue a reload), but --test-logs
-				// would fail (and not reload).
-				doLogTest(servers, globalCollectionOpts, logger)
-			}
-
-			reloadOkay = allFullSuccessful && allActivitySuccessful
-			if reloadOkay {
-				// in a dry run, we will not actually have URLs; avoid this output in that case
-				var hasURLs bool
+		wg.Add(1)
+		testRunSuccess = make(chan bool)
+		go func() {
+			if globalCollectionOpts.TestReport != "" {
+				runner.RunTestReport(ctx, servers, globalCollectionOpts, logger)
+				testRunSuccess <- true
+			} else if globalCollectionOpts.TestExplain {
+				success := true
 				for _, server := range servers {
-					if server.PGAnalyzeURL != "" {
-						hasURLs = true
-						break
+					prefixedLogger := logger.WithPrefix(server.Config.SectionName)
+					err := runner.EmitTestExplain(ctx, server, globalCollectionOpts, prefixedLogger)
+					if err != nil {
+						prefixedLogger.PrintError("Failed to run test explain: %s", err)
+						success = false
 					}
 				}
-				if hasURLs {
-					fmt.Fprintln(os.Stderr)
-					fmt.Fprintln(os.Stderr, "Test successful. View servers in pganalyze:")
+				testRunSuccess <- success
+			} else if globalCollectionOpts.TestRunLogs {
+				success := doLogTest(ctx, servers, globalCollectionOpts, logger)
+				testRunSuccess <- success
+			} else {
+				var allFullSuccessful bool
+				var allActivitySuccessful bool
+				allFullSuccessful = runner.CollectAllServers(ctx, servers, globalCollectionOpts, logger)
+				if hasAnyActivityEnabled && ctx.Err() == nil {
+					allActivitySuccessful = runner.CollectActivityFromAllServers(ctx, servers, globalCollectionOpts, logger)
+				} else {
+					allActivitySuccessful = true
```

Comment:
Is this still right? We set `allActivitySuccessful = true` if `ctx.Err() != nil`?